feat(ai-gateway): Update load balancing capabilities documentation for AI Gateway 2.0 (#5308)
Kong/developer.konghq.com: app/_ai_gateway_entities/ai-model.md

  - text: Consumer Group entity
    url: /ai-gateway/entities/ai-consumer-group/
faqs:
  - q: What's the difference between a Model entity and the `model` field in a Policy configuration?
    a: |

>       A Model entity is the first-class {{site.ai_gateway}} entity you declare through the {{site.konnect_short_name}} API, UI, or decK.

Reviewer: deck?

Author: I'll remove it, thanks for catching this one

* `$(uri_captures.path_parameter_name)`: the value of a captured URI path parameter.
* `$(query_params.query_parameter_name)`: the value of a query string parameter.

> For examples on using templating, consult the {{site.ai_gateway}} documentation and API reference.

Reviewer: ```suggestion
For examples of using templating, consult the {{site.ai_gateway}} documentation and API reference.
```

A Model declares the Policies it uses through its `policies` field. Each entry is a string that references a Policy by name or ID. {{site.konnect_short_name}} resolves these references against Policies created at `/v1/ai-gateways/{aiGatewayId}/policies`.

You can attach multiple Policies to a single Model. Each Policy is applied independently, so attaching the same Policy type twice with different configurations creates two separate instances.

Not every Policy type is valid as a Model attachment.

Policies attached to a Model are not deleted when the Model is deleted; only the Model's reference is removed.

For further information, see the [Policy entity](/ai-gateway/entities/ai-policy/) reference.

### Plugin priority and Policy execution order

A Policy attached to a Model runs on the Service of the Model's derived primitives. That Policy runs at the [priority](/gateway/entities/plugin/#plugin-priority) determined by its type, which affects when it executes relative to other Policies on the request.

> Model routing itself executes at a specific point in the request pipeline. Policies whose types run before that point (higher priority) execute before the Model is resolved. Authentication Policies (such as OpenID Connect) fall into this category. They gate access correctly because routing to the Model's generated Service already occurred, but model-level identity details (provider and target model) are not available until after Model resolution.

Reviewer: ```suggestion
Model routing executes at a specific point in the request pipeline. Policies have different priorities that determine when they run.  Higher priority Policies types may run before the Model routing is resolved. Authentication Policies (such as OpenID Connect) fall into this category. They gate access correctly because routing to the Model's generated Service already occurred, but model-level identity details (provider and target model) are not available until after Model resolution.
```
